newlib/libc/tinystdio/fdevopen.c

```diff
@@ -83,8 +83,12 @@ static int
 fdevclose(FILE *f)
 {
 	int ret = 0;
+
+	__flockfile(f);
 	if  (f->flush)
 		ret = (*f->flush)(f);
+	__flockfile_close(f);
```

Comment:
I think this needs a call to `__funlockfile` before the call to `__flockfile_close`.

Reply:
I resolved it by adding `__lock_release` call into `__flockfile_close` internals. Like it was done for `__bufio_lock_close`